source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc

```diff
@@ -0,0 +1,217 @@
+#include "source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.h"
+
+#include "envoy/extensions/network/dns_resolver/getaddrinfo/v3/getaddrinfo_dns_resolver.pb.h"
+#include "envoy/network/dns_resolver.h"
+
+#include "source/common/api/os_sys_calls_impl.h"
+#include "source/common/network/address_impl.h"
+
+namespace Envoy {
+namespace Network {
+
+// This resolver uses getaddrinfo() on a dedicated resolution thread. Thus, it is only suitable
+// currently for relatively low rate resolutions. In the future a thread pool could be added if
+// desired.
```

Comment:
Should have a note in the docs as well, yes?

Reply:
Yes will do.